### PR TITLE
decl_module! macro: use 'frame_system' instead of `system` as default ident

### PR DESCRIPTION
**Pull request by shaopengw**
`decl_module!` macro uses `system` as default `ident` for `frame-system` module. To use it in runtime modules, `frame-system` always needs to be renamed as `system`. (Adding a `system = frame_system` bound works as well, but I didn't find a use case in FRAME modules).

This PR makes `frame_system` as the default `ident`, so most `frame_system -> system` renaming cases could be avoided.

I've also removed unused renaming with this change in some FRAME modules, where the renaming is only to satisfy `decl_module!`. In other FRAME modules where `system` is actually used, the renaming are kept, for I'm not sure if the renaming is only for `decl_module!`, or intended to make it shorter as `system`.

polkadot companion: paritytech/polkadot#1314

### CODE REVIEW

**Comment by thiolliere** [comment]
>  In other FRAME modules where system is actually used, the renaming are kept, for I'm not sure if the renaming is only for decl_module!, or intended to make it shorter as system.

I think the renaming was for decl_module and some later change made use of it. in general I would accept a PR with no renaming at all in substrate, but I'm also ok with this PR at its current state

**Comment by gavofyork** [comment]
needs a polkadot companion.

File: frame/sudo/src/mock.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ use sp_runtime::{Perbill, traits::{BlakeTwo256, IdentityLookup}, testing::Header
 use sp_io;
 use crate as sudo;
 use frame_support::traits::Filter;
+use frame_system as system;
```
Comment (bkchr) [inline frame/sudo/src/mock.rs:32]: Why do you rename it here?

File: frame/utility/src/tests.rs [hunk 2]
```diff
@@ -25,6 +25,7 @@ use frame_support::{
 	assert_ok, assert_noop, impl_outer_origin, parameter_types, impl_outer_dispatch,
 	weights::Weight, impl_outer_event, dispatch::DispatchError, traits::Filter, storage,
 };
+use frame_system as system;
```
Comment (bkchr) [inline frame/utility/src/tests.rs:28]: Also here.

File: frame/sudo/src/mock.rs (hunk 1 above)
Comment (shaopengw) [reply frame/sudo/src/mock.rs:32]: `system` from `use super::*;` is used in this mock mod, and after the renaming in `fram/sudo/src/lib.rs` got deleted, there is no `system` in mock any more.

I just double checked and it seems `system` and `frame_system` are both used here. Will change `system` to `frame_system` to make it consistent.

File: frame/utility/src/tests.rs (hunk 2 above)
Comment (shaopengw) [reply frame/utility/src/tests.rs:28]: Same to the reason in `frame/sudo/src/mock.rs`.

Mixed `frame_system` and `system` also happens here, will update.

**Comment by shaopengw** [comment]
I've made a Polkadot companion PR, but not sure if I missed something since the companion build is still in a failed state. Or it will be left as failed, and I could fix the Polkadot build failure later after this PR got merge?

**Comment by gavofyork** [comment]
polkadot companion not building.

**Comment by shaopengw** [comment]
> polkadot companion not building.

Previous failures were because of not linking companion PR in the description. Just updated and now it builds with the right branch.

**Comment by shaopengw** [comment]
Polkadot companion still not building, but it looks like not caused by this PR https://gitlab.parity.io/parity/substrate/-/jobs/579007

The lastest companion build failure in master branch has the same errors https://gitlab.parity.io/parity/substrate/-/jobs/578276

**Comment by gavofyork** [comment]
A merge with master should help...

**Comment by shaopengw** [comment]
It's green now.

**Comment by xlc** [comment]
bot merge

**Pull request merged**